c/planarityTestGraphFunctionality.c

```diff
@@ -111,3 +212,130 @@ int transformString(graphP theGraph, char *inputStr)
+*/
+
+int testAllGraphs(graphP theGraph, char command, char *inputStr, strOrFileP testOutput)
+{
+	int exitCode = OK;
+
+	char *errorStr = NULL;
+
+	graphP copyOfOrigGraph = NULL;
+	int embedFlags = GetEmbedFlags(command);
+	int numGraphsRead = 0, numOK = 0, numNONEMBEDDABLE = 0;
+
+	G6ReadIterator *pG6ReadIterator = NULL;
+	exitCode = allocateG6ReadIterator(&pG6ReadIterator, theGraph);
+
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to allocate G6ReadIterator.\n");
+		return exitCode;
+	}
+
+	exitCode = beginG6ReadIterationFromG6String(pG6ReadIterator, inputStr);
+
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to begin .g6 read iteration.\n");
+
+		exitCode = freeG6ReadIterator(&pG6ReadIterator);
+
+		if (exitCode != OK)
+			ErrorMessage("Unable to free G6ReadIterator.\n");
+
+		return exitCode;
+	}
+
+	AttachAlgorithm(pG6ReadIterator->currGraph, command);
+
+	copyOfOrigGraph = gp_New();
+	if (copyOfOrigGraph == NULL)
+	{
+		ErrorMessage("Unable to allocate graph to store copy of original graph before embedding.\n");
+		return NOTOK;
+	}
+
+	exitCode = gp_InitGraph(copyOfOrigGraph, pG6ReadIterator->graphOrder);
+	if (exitCode != OK)
+	{
+		ErrorMessage("Unable to initialize graph datastructure to store copy of original graph before embedding.\n");
+		gp_Free(&copyOfOrigGraph);
+		freeG6ReadIterator(&pG6ReadIterator);
+	}
+
+	AttachAlgorithm(copyOfOrigGraph, command);
+
+	while (true)
+	{
+		exitCode = readGraphUsingG6ReadIterator(pG6ReadIterator);
+
+		if (exitCode != OK)
+		{
+			errorStr = "Unable to read graph on line %d from .g6 read iterator.\n";
+			sprintf(Line, errorStr, pG6ReadIterator->numGraphsRead + 1);
+			ErrorMessage(Line);
+			break;
+		}
+
+		if (pG6ReadIterator->currGraph == NULL)
+			break;
+		
+		gp_CopyGraph(copyOfOrigGraph, pG6ReadIterator->currGraph);
+
+		exitCode = gp_Embed(pG6ReadIterator->currGraph, embedFlags);
+
+		if (gp_TestEmbedResultIntegrity(pG6ReadIterator->currGraph, copyOfOrigGraph, exitCode) != exitCode)
+			exitCode = NOTOK;
+		
+		if (exitCode == OK)
+			numOK++;
+		else if (exitCode == NONEMBEDDABLE)
+			numNONEMBEDDABLE++;
+		else
+		{
+			errorStr = "Error applying algorithm '%c' to graph on line %d.\n";
+			sprintf(Line, errorStr, command, pG6ReadIterator->numGraphsRead + 1);
+			ErrorMessage(Line);
+			break;
+		}
+
+		gp_ReinitializeGraph(copyOfOrigGraph);
+	}
+
+	if (exitCode == OK || exitCode == NONEMBEDDABLE)
+	{
+		// pG6ReadIterator->numGraphsRead is only incremented after successfully decoding
+		numGraphsRead = pG6ReadIterator->numGraphsRead;
+		char *resultsStr = (char *) malloc((3 +_getNumCharsToReprInt(numGraphsRead) +
+											1 + _getNumCharsToReprInt(numOK) +
+											1 + _getNumCharsToReprInt(numNONEMBEDDABLE) + 3) * sizeof(char));
+		sprintf(resultsStr, "-%c %d %d %d\n", command, numGraphsRead, numOK, numNONEMBEDDABLE);
+		sf_fputs(resultsStr, testOutput);
+		free(resultsStr);
+		resultsStr = NULL;
+	}
+
+	exitCode = endG6ReadIteration(pG6ReadIterator);
```

Comment:
The following two exitCode reassignments will also possibly hide a failure that occurred in the core loop. Better to just say `if (endG6ReadInteration(...) != OK) error message` and same for `freeG6ReadIterator()`.